addons/hr/tests/test_self_user_access.py

```diff
@@ -216,3 +216,15 @@ def testWriteOtherUserEmployee(self):
     def testSearchUserEMployee(self):
         # Searching user based on employee_id field should not raise bad query error
         self.env['res.users'].with_user(self.richard).search([('employee_id', 'ilike', 'Hubert')])
+
+    def test_onchange_readable_fields_with_no_access(self):
+        """
+            The objectif is to test that the onchange logic takes account of `SELF_READABLE_FIELDS`.
+
+            The view contains fields in `SELF_READABLE_FIELDS` (example: `private_street`).
+            Even if the user does not have read access to the employee,
+            it should not cause an access error if these fields are in `SELF_READABLE_FIELDS`.
+        """
+        self.env.ref("base.lang_fr").active = True
+        user_form = Form(self.richard.with_user(self.richard), view=self.env.ref('hr.res_users_view_form_profile'))
+        user_form.lang = "fr_FR" # Trigger an onchange
```

Comment:
You can use the `with` form of it, and `view` can be a string
```suggestion
        with Form(self.richard.with_user(self.richard), view='hr.res_users_view_form_profile') as form:
            # triggering an onchange should not trigger some access error
            form.lang = "fr_FR"
```